internal/compiler/passes/lower_property_to_element.rs

```diff
@@ -10,16 +10,23 @@ use std::rc::Rc;
 use crate::diagnostics::BuildDiagnostics;
 use crate::expression_tree::{BindingExpression, NamedReference};
 use crate::langtype::Type;
-use crate::object_tree::{self, BindingsMap, Component, Element, ElementRc};
+use crate::object_tree::{self, Component, Element, ElementRc};
 use crate::typeregister::TypeRegister;
 
+// If any element in `component` declares a binding to the first property in `properties`, then a new
+// element of type `element_name` is created, injected as a parent to the element and bindings to all
+// remaining properties in `properties` are mapped. This way for example ["rotation-angle", "rotation-origin-x", "rotation-origin-y"]
+// creates a `Rotate` element when `rotation-angle` is used and any optional `rotation-origin-*` bindings are redirected to the
+// `Rotate` element.
 pub(crate) fn lower_property_to_element(
     component: &Rc<Component>,
-    property_name: &str,
+    properties: &[(&str, Type)],
     element_name: &str,
     type_register: &TypeRegister,
     diag: &mut BuildDiagnostics,
 ) {
+    let property_name = properties[0].0;
```

Comment:
This is also a bit magical.  

Maybe the signature could be:   

```rust
pub(crate) fn lower_property_to_element(
    component: &Rc<Component>,
    property_name: &str,
    extra_properties: &[&str],
    element_name: &str,
    type_register: &TypeRegister,
 ```

Even if that means one can't pass RESERVED_ROTATION_PROPERTIES directly, but need a map().

Reply:
Done. A map was not sufficient, a collect was required :(. I've changed it to take an impl Iterator for the extra properties in a follow-up commit. Let me know if that's ok :)